client/components/sites-dropdown/index.jsx

```diff
@@ -101,3 +102,10 @@ export default class SitesDropdown extends PureComponent {
 		);
 	}
 }
+
+export default connect( ( state ) => {
+	return {
+		getSite: getSite.bind( null, state ),
```

Comment:
Ahem 😉 

In all seriousness: There aren't that many consumers. Ideally we just return the ID and let 'em do their thing.